examples/video-chat/components/VideosGrid.tsx

```diff
@@ -1,119 +1,100 @@
 import {
   Track,
   VideoRendererView,
-  setTargetTrackEncoding,
   Metadata,
+  Peer,
 } from '@fishjam-cloud/react-native-client';
 import React from 'react';
-import { Dimensions, StyleSheet, View } from 'react-native';
-import Animated, {
-  FadeInDown,
-  LinearTransition,
-} from 'react-native-reanimated';
+import { FlatList, StyleSheet, View } from 'react-native';
 
-import LetterButton from './LetterButton';
 import { roomScreenLabels } from '../types/ComponentLabels';
+import { BrandColors } from '../utils/Colors';
+import Typo from './Typo';
 
 type Props = {
-  tracks: Track<Metadata>[];
+  tracks: GridTrack[];
+};
+
+type GridTrack = Track<Metadata> & {
+  isLocal: boolean;
+  userName: string | undefined;
 };
 
-const { width } = Dimensions.get('window');
 const { VIDEO_CELL } = roomScreenLabels;
-const AnimatedVideoRenderer =
-  Animated.createAnimatedComponent(VideoRendererView);
 
-export default function VideosGrid({ tracks }: Props) {
-  const videoWidth = (width - 40) / 2;
+export function parsePeersToTracks(
+  peers: Peer<Metadata, Metadata, Metadata>[],
+): GridTrack[] {
+  return peers
+    .sort((peer) => (peer.isLocal ? -1 : 1))
+    .flatMap((peer) =>
+      peer.tracks
+        .map((track) => ({
+          ...track,
+          isLocal: peer.isLocal,
+          userName: peer.metadata?.name,
+        }))
+        .filter(
+          (t) => t.metadata.type !== 'audio' && (t.metadata.active ?? true),
+        ),
+    );
+}
 
+export default function VideosGrid({ tracks }: Props) {
   return (
-    <View
-      style={
-        tracks.length > 3 ? styles.videosContainer2 : styles.videosContainer1
-      }>
-      {tracks.map((v, idx) => (
-        <Animated.View
+    <FlatList<GridTrack>
+      data={tracks}
+      renderItem={({ item: track, index: idx }) => (
+        <View
           accessibilityLabel={VIDEO_CELL + idx}
-          entering={FadeInDown.duration(200)}
-          layout={LinearTransition.duration(150)}
-          style={
-            tracks.length > 3
-              ? [styles.video2, { width: videoWidth, height: videoWidth }]
-              : [styles.video1, { maxWidth: width - 20 }]
-          }
-          key={v.id}>
-          <AnimatedVideoRenderer
-            trackId={v.id}
-            entering={FadeInDown.duration(200)}
-            layout={LinearTransition.duration(150)}
-            style={styles.animatedView}
+          style={[
+            styles.video,
+            {
+              backgroundColor: track.isLocal
+                ? BrandColors.yellow100
+                : BrandColors.darkBlue60,
```

Comment:
This colors can be considered ugly. I'm open for proposals here.

Reply:
It can be a feature for "FIT" property.
My proposal is to add background view as "FILL" and then blur it. Many aps use that patter for fitting videos/images. What do you think about it?

Reply:
For now I would prefer to keep this app simple, to make sure that our core functionalities works. Bluring/animations - it is nice, but we don't have time to support it now :(